specs/data_structures.md

```diff
@@ -29,45 +29,54 @@ Data Structures
 
 ## Header
 
-| name                 | type                      | description |
-| -------------------- | ------------------------- | ----------- |
-| `version`            | [Version](#version)       |             |
-| `chainID`            | `uint64`                  |             |
-| `height`             | `uint64`                  |             |
-| `time`               | [Time](#time)             |             |
-| `lastBlockID`        | [BlockID](#blockid)       |             |
-| `lastCommitHash`     | [HashDigest](#hashdigest) |             |
-| `dataHash`           | [HashDigest](#hashdigest) |             |
-| `validatorsHash`     | [HashDigest](#hashdigest) |             |
-| `nextValidatorsHash` | [HashDigest](#hashdigest) |             |
-| `consensusHash`      | [HashDigest](#hashdigest) |             |
-| `appHash`            | [HashDigest](#hashdigest) |             |
-| `lastResultsHash`    | [HashDigest](#hashdigest) |             |
-| `evidenceHash`       | [HashDigest](#hashdigest) |             |
-| `proposerAddress`    | [Address](#address)       |             |
+Block header, which is fully downloaded by both full clients and light clients.
+
+| name                 | type                        | description                                                 |
+| -------------------- | --------------------------- | ----------------------------------------------------------- |
+| `version`            | `uint64`                    | Version of the LazyLedger chain.                            |
+| `chainID`            | `uint64`                    | Chain ID. Each fork assigns itself a (unique) ID.           |
+| `height`             | `uint64`                    | Block height. The genesis block is at height `1`.           |
+| `time`               | [Time](#time)               | Timestamp of this block.                                    |
+| `lastBlockID`        | [BlockID](#blockid)         | Previous block's ID.                                        |
+| `lastCommitRoot`     | [HashDigest](#hashdigest)   | Previous block's Tendermint commit root.                    |
+| `dataRoot`           | [HashDigest](#hashdigest)   | Message data root.                                          |
+| `validatorsRoot`     | [HashDigest](#hashdigest)   | Validator set root for this block.                          |
+| `nextValidatorsHash` | [HashDigest](#hashdigest)   | Root of the next block's validator set.                     |
+| `consensusHash`      | [HashDigest](#hashdigest)   | Consensus parameters for this block.                        |
+| `appHash`            | [HashDigest](#hashdigest)   |                                                             |
+| `lastResultsHash`    | [HashDigest](#hashdigest)   |                                                             |
+| `evidenceRoot`       | [HashDigest](#hashdigest)   | Evidence data root.                                         |
```

Comment:
Does the slashing evidence need data availability? If so, they should be under the erasure coded data root.

Reply:
Actually, maybe not? Evidence is needed to slash validators, which means it affects the validator set, which means it needs to be fully downloaded by both full and light nodes I think.

Reply:
After some deliberation, putting the evidence data under the data availability root is a good idea because it enables super-light clients that don't actually process validator set changes. Refactoring in #12.